applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -90,11 +90,8 @@ ModelPart& EmbeddedSkinVisualizationProcess::CreateAndPrepareVisualizationModelP
         r_visualization_variables_list.Add(r_var);
     }
 
-    // Set the origin model part as temporary communicator
-    // This is required to perform all the IsDistributed checks that appear before creating the visualization mesh
-    // Note that these checks might be required outside this process (i.e. in the creation of the visualization mesh output)
-    // This will be updated by a proper one by the ParallelFillCommunicator after the creation of the visualization entities
-    r_visualization_model_part.SetCommunicator(r_origin_model_part.pGetCommunicator());
```

Comment:
so basically you cannot resue the same `Communicator` as the visualization MP contains more nodes right?

Then why not use the existing interface to create a new `Communicator`:
~~~c++
r_visualization_model_part.SetCommunicator(
    r_origin_model_part.GetCommunicator().Create(
        r_origin_model_part.GetCommunicator().GetDataCommunicator()));
~~~

Reply:
Genau. The visualization model part nodes depend on the intersection pattern.

There is no special reason to not use your proposal  as the result would be exactly the same. The unique thing is that the data communicator instance would be unique in your case.

Having said this, I don't know if this might be helpful or a problem if we need to eventually debug something. Aside of this, I think that using the `ParallelEnvironment` would be more clear.

Reply:
> The unique thing is that the data communicator instance would be unique in your case.

this is essentially what we should enforce, otherwise we might have bad deadlocks once we use `MPI_COMM`s that are not `MPI_COMM_WORLD`

adding Interfaces that enable to go around this should not be added IMO, unless we really really need them, then we can take a look again

Reply:
I can foresee a usage of a not `MPI_COMM_WORLD` MPI_COMM. Are we planning to do that? If this is not the case, I think adding interfaces that make things easier is not bad. If we plan to do that, I've no problem on using your proposal.

Reply:
I have a development branch for using it in the CoSim (e.g. FSI where fluid uses all MPI core but structure only uses a part of them)
@pooyan-dadvand and I also need this for a project with Altair

In any case I don't think this is restricted to CoSim cases, I can imagine that it can also be used for other things

Reply:
The issue with adding simple interfaces for complex things such as this can and most probably will haunt us eventually
I am observing the usage of the default `DataCommunicator` for a while already and I don't like what I see :D

IMO we could even thing of removing those simple interfaces as they make misuse too easy

In any case lets discuss it also with @KratosMultiphysics/technical-committee

Reply:
> I have a development branch for using it in the CoSim (e.g. FSI where fluid uses all MPI core but structure only uses a part of them)
> @pooyan-dadvand and I also need this for a project with Altair
> 
> In any case I don't think this is restricted to CoSim cases, I can imagine that it can also be used for other things

But wouldn't this be something related with the partitioning rather than with the communicator itself?

Reply:
If the nodes in the two model parts are different, then I don't think this is a good idea to have the same communicator. (Consider that communicator would have the list of the ghost nodes which is valid only for one of the model parts.

Reply:
Solved. Using the current API as makes no difference from the utility perspective.